src/UserAdmin/src/Smart.FA.Catalog.Web/Authentication/Handlers/UserAdminAuthenticationHandler.cs

```diff
@@ -159,12 +165,25 @@ private async Task SetUserIdentityAsync(Trainer trainer)
     {
         // Sets the data for the IUserIdentity service.
         var isAdmin = await _mediator.Send(new IsSuperUserQuery(trainer.Id));
-        Context.User = new GenericPrincipal(new CustomIdentity(trainer), roles: isAdmin ? new[] { "SuperUser" } : null);
-
+        var roles = new List<string>();
+        roles.AddIf(() => isAdmin, Roles.SuperUser);
+        roles.AddIf(() => IsSocialMember(trainer.Identity.UserId), Roles.SocialMember);
+        var connectedUser = _adminBehindUser is null
+            ? null
+            : new ConnectedUser(_adminBehindUser.UserId!, _adminBehindUser.Email!, Name.Create(_adminBehindUser.FirstName!, _adminBehindUser.LastName!).Value);
+        Context.User = new GenericPrincipal(new CustomIdentity(trainer, connectedUser), roles.ToArray());
         // Updates the first name, last name and email address of the current trainer if they changed for any reason.
         // If anything goes wrong an exception will be thrown and stops execution of the HTTP request.
         await _mediator.Send(new UpdateTrainerIdentityCommand(trainer.Id, _firstName, _lastName, _email));
     }
+
+    /// <summary>
+    /// Check if the connected user is not impersonating a social member and is directly connected as a Permanent member.
+    /// The "AD" prefix before the user id is a convention adopted between the app to indicate permanent member access.
+    /// </summary>
+    /// <param name="userId">The user id of the connected user</param>
+    /// <returns></returns>
+    private static bool IsSocialMember(string userId) => !Regex.Match(userId, "^AD").Success;
```

Comment:
Why complicate the code with `Regex`?
```suggestion
    private static bool IsSocialMember(string userId) => userId.ToUpper().StartsWith("AD");
```

Reply:
Yes, sure why not (without the ToUpper though)!

Reply:
Ther's an overload StringComparaison.IgnoreCase something. No need to use the ToUpper